api/src/services/payload.ts

```diff
@@ -670,17 +670,38 @@ export class PayloadService {
 				if (error) throw new InvalidPayloadException(`Invalid one-to-many update structure: ${error.message}`);
 
 				if (alterations.create) {
-					await itemsService.createMany(
-						alterations.create.map((item) => ({
-							...item,
-							[relation.field]: parent || payload[currentPrimaryKeyField],
-						})),
-						{
-							onRevisionCreate: (pk) => revisions.push(pk),
-							bypassEmitAction: (params) => nestedActionEvents.push(params),
-							emitEvents: opts?.emitEvents,
+					const sortField = relation.meta.sort_field;
+
+					const processCreateItem = async (item: Record<string, any>, index: number) => {
+						const record = cloneDeep(item);
+
+						// add sort field value if it is not supplied in the item
+						if (sortField !== null && parent !== null && record[sortField] === undefined) {
+							const highestOrderNumber: Record<string, any> | undefined = await this.knex
+								.select(sortField)
+								.from(relation.collection)
+								.where({ [relation.field]: parent })
+								.whereNotNull(sortField)
+								.orderBy(sortField, 'desc')
+								.first();
```

Comment:
This does a new SQL query for every item in the create array, that's a pretty large amount of overhead. We should instead fetch the current highest value from the DB once, and keep incrementing in JS when looping over instead.

Reply:
implemented 👍